### DISCUSSION TITLE
Linked user ID's between tables error

### DISCUSSION THREAD

**Discussion by radu-mee** [supabase/supabase#27384] [Questions]
Hello everyone, I don't know if this is a bug or not, but this is what is happening. I created an additional users tablein the public schema, that I have linked to the users table in the auth schema in order to capture more data besides the email and password, such as first name, last name.

I have linked those 2 tables through the user ID of type UUID. But when I try to register a test user, it gives me an error 
"Error: null value in column "id" of relation "users" violates not-null constraint at Object.insert"

And when I look at the foreign key relationships details in Supabase I have this message even though both ID's are in fact of type UUID: 
"Column types do not match
The following columns cannot be referenced as they are not of the same type:
id (uuid) and id()
"
![11111](https://github.com/supabase/supabase/assets/172914733/4dbef23d-8d68-4477-a1d0-d469e2a4d092)
![12311131](https://github.com/supabase/supabase/assets/172914733/4e1fc5cd-cf7f-413d-8761-96be8741d93c)
![412341234](https://github.com/supabase/supabase/assets/172914733/c224f373-7814-4d64-95d5-0ad94102f336)

**Comment by GaryAustin1** [comment]
Did you change your id type at some point?  I just verified it is working. 
![image](https://github.com/supabase/supabase/assets/54564956/840dce4b-e8be-44bb-9f02-7a40518b0628)


As far as your create user error.  Look in the Auth and Postgres logs for a specific error.  Not sure if both of these are related or your trigger function is not

**Comment by filmackay** [comment]
I am getting the same issue here - exact same symptoms. I didn't change the type at any point..

**Reply by mikelong10** [reply]
I'm also seeing the same thing, also didn't change my id type ever

**Comment by vibecodehater** [comment]
hi, i was getting this error and finally fixed it. all i had to do was delete the table and create it again, and set type 'uuid' - 'gen_random_uuid()' from the beginning. it seems like everything crashes when you try to make changes to an already existing table
